x/pocketcore/types/session.go

```diff
@@ -124,17 +125,21 @@ func NewSessionNodes(sessionCtx, ctx sdk.Ctx, keeper PosKeeper, chain string, se
 		sessionKey = Hash(sessionKey)
 		// get the node from the array
 		n := nodesAddrs[index.Int64()]
-		//if we already have seen this address we continue as it's either on the list or discarded
+		// if we already have seen this address we continue as it's either on the list or discarded
 		if _, ok := m[n.String()]; ok {
 			continue
 		}
-		//add the node address to the map
+		// add the node address to the map
 		m[n.String()] = struct{}{}
 
 		// cross check the node from the `new` or `end` world state
 		node = keeper.Validator(ctx, n)
-		// if not found or jailed, don't add to session and continue
-		if node == nil || node.IsJailed() || !NodeHasChain(chain, node) || sessionNodes.Contains(node.GetAddress()) {
+		// if not found or jailed or staked chains is too large, don't add to session and continue
+		if node == nil ||
+			node.IsJailed() ||
+			!NodeHasChain(chain, node) ||
+			sessionNodes.Contains(node.GetAddress()) ||
+			!keeper.ValidatorHasLessOrEqualMaxChains(ctx, node) {
```

Comment:
This is a consensus breaking change.  Can you add a feature flag in codec/codec.go and apply this new condition only if the height >= the activation height?

Reply:
How should I determine the height of activation?

Reply:
Look here for some examples: https://github.com/pokt-network/pocket-core/blob/staging/codec/codec.go#L287

Reply:
You don't need to determine the block height.  You just define a feature name as a string.  The height is determined when we submit `gov enable` command to activate the feature.  Here's a PR defining a new feature: https://github.com/pokt-network/pocket-core/pull/1534/files#diff-8b594c387517838ab44fd9d9f7a5f8c1e3efa32486a1ba4b923858e4e6538955